### PR TITLE
bugfix/#5454-show-popup-on-correct-condition

### PR DESCRIPTION
**Pull request by KKatiaa**
Bugfix#5454 Pop up with the full data does not appear correctly when admin clicks on the string in the Orders table

### CODE REVIEW

**Review by atsa21 — APPROVED** [review]

File: src/app/ubs/ubs-admin/components/ubs-admin-table/table-cell-readonly/table-cell-readonly.component.ts [hunk 1]
```diff
@@ -28,9 +29,26 @@ export class TableCellReadonlyComponent implements OnInit, OnChanges {
...
   }
 
-  isTooltipEnabled(title: any, textContainerWidth: number = 112): boolean {
-    const textWidth = document.createElement('canvas').getContext('2d').measureText(title).width;
+  showTooltip(event, tooltip) {
+    event.stopImmediatePropagation();
 
-    return textContainerWidth < textWidth;
+    const lengthStr = event.toElement?.innerText.split('').length;
+    if (lengthStr > 50) {
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/ubs-admin-table/table-cell-readonly/table-cell-readonly.component.ts]: move this value (50) in default parameter for method

File: src/app/ubs/ubs-admin/components/ubs-admin-table/table-cell-readonly/table-cell-readonly.component.ts [hunk 2]
```diff
@@ -28,9 +29,26 @@ export class TableCellReadonlyComponent implements OnInit, OnChanges {
...
+  }
+
+  calculateTextWidth(event, tooltip): void {
+    const textContainerWidth = event.toElement.offsetWidth;
+    const canvas = document.createElement('canvas');
+    const context = canvas.getContext('2d');
+    context.font = '12px Lato, sans-serif';
+    const textWidth = Math.round(context.measureText(event.toElement.innerText).width);
+
+    if (textContainerWidth < textWidth || Math.abs(textContainerWidth - textWidth) < 40) {
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/ubs-admin-table/table-cell-readonly/table-cell-readonly.component.ts]: the same

File: src/app/ubs/ubs-admin/components/ubs-admin-table/table-cell-readonly/table-cell-readonly.component.ts [hunk 3]
```diff
@@ -28,9 +29,26 @@ export class TableCellReadonlyComponent implements OnInit, OnChanges {
...
-    return textContainerWidth < textWidth;
+    const lengthStr = event.toElement?.innerText.split('').length;
+    if (lengthStr > 50) {
+      tooltip.toggle();
+    }
+
+    event.type === MouseEvents.MouseEnter ? this.calculateTextWidth(event, tooltip) : tooltip.hide();
+  }
+
+  calculateTextWidth(event, tooltip): void {
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/ubs-admin-table/table-cell-readonly/table-cell-readonly.component.ts]: add types for parameters

File: src/app/ubs/ubs-admin/components/ubs-admin-table/table-cell-readonly/table-cell-readonly.component.ts [hunk 4]
```diff
@@ -28,9 +29,26 @@ export class TableCellReadonlyComponent implements OnInit, OnChanges {
         : this.title;
   }
 
-  isTooltipEnabled(title: any, textContainerWidth: number = 112): boolean {
-    const textWidth = document.createElement('canvas').getContext('2d').measureText(title).width;
+  showTooltip(event, tooltip) {
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/ubs-admin-table/table-cell-readonly/table-cell-readonly.component.ts]: add types for parameters and result

**Review by NadiiaStoiko — APPROVED** [review]

**Review by atsa21 — APPROVED** [review]

**Review by AlexW8719 — APPROVED** [review]

**Review by katia9261 — APPROVED** [review]

**Review by hnativlyubomyr — APPROVED** [review]

**Pull request merged**